esp/services/ws_decoupledlogging/ws_decoupledloggingservice.hpp

```diff
@@ -33,13 +33,18 @@ class CWSDecoupledLogSoapBindingEx : public CWSDecoupledLogSoapBinding
         : CWSDecoupledLogSoapBinding(cfg, bindname, procname, level) { };
 };
 
+struct WSDecoupledLogAgentGroup
+{
+    StringAttr name;
+    StringAttr tankFileDir, tankFileMask;
+    std::vector<IUpdateLogThread*>  loggingAgentThreads;
```

Comment:
Should the threads be either Owned or Linked? If the group logically owns them, this definition requires a destructor to release them. If the group do not logically own them, it should link to them to ensure they aren't destroyed by the owner.

Yes, the lifetime of the group  is the lifetime of the service, which is expected to be the lifetime of the process. It's just that a very small change can avoid adding to the noise generated by tools like valgrind when they report "leaks" that really aren't.